TextMiningFinalProject.py

```diff
@@ -0,0 +1,199 @@
+    y_pos = np.arange(len(objects))
+    performance = [.096,.11,.1,.159]
+    plt.bar(y_pos, performance, align='center', alpha=0.5)
+    plt.xticks(y_pos, objects)
+    plt.ylabel('Positive Sentiment Ratio')
+    plt.title('Positive Sentiment')
+    plt.show()
+    plt.savefig("sentiment_pos.png")
+
+#same as well for negative sentiment
+def making_graphs_neg():
+    objects = ('Communist Manifest','Constitution','Trotsky','Utilitarianism')
+    y_pos = np.arange(len(objects))
+    performance = [.08,.054,.11,.094]
+    plt.bar(y_pos, performance, align='center', alpha=0.5)
+    plt.xticks(y_pos, objects)
+    plt.ylabel('Negative Sentiment Ratio')
+    plt.title('Negative Sentiment')
+    plt.show()
+    plt.savefig("sentiment_neg.png")
+
+    
+
+#making_graphs_pos()
+#making_graphs_neg()
+#making_graphs_neutral()
+
+"""
+Results of the sentiment analysis:
+{'compound': 0.9998, 'neu': 0.824, 'neg': 0.08, 'pos': 0.096}
+{'compound': 0.9999, 'neu': 0.836, 'neg': 0.054, 'pos': 0.11}
+{'compound': -1.0, 'neu': 0.79, 'neg': 0.11, 'pos': 0.1}
+{'compound': 1.0, 'neu': 0.747, 'neg': 0.094, 'pos': 0.159}"""
+
+#processing each of the files and making a histogram out of them
+def process_file(name_of_file):
+    hist = dict()
+    open_file = open(name_of_file, 'rb')
+    for l in open_file:
+        process_line(l,hist)
+    return hist
+
+def process_line(line,hist):
+    #line = line.replace('-', ' ')
+    for word in line.split():
+        #word = word.strip(string.punctuation + string.whitespace) #TODO y no work
+        word = word.lower()
+        hist[word]=hist.get(word,0)+1
+
+#processing each of the pickled texts
+constitution_hist = process_file('constitution.pickle')
+communist_manifesto_hist = process_file('communist_manifesto.pickle')
+trotsky_hist = process_file('trotsky.pickle')
+utilitarianism_hist = process_file('utilitarianism.pickle')
+
+#eliminates any word shorter than b and then determines the most common words of a dictionary (n amount)
+def top_words(dictionary,n,b):
+    """
+    >>> top_words({'cat':1,'dog':1,'elephant':5,'soul':10,'jazz':20},3,2)
+    [(5, 'elephant'), (10, 'soul'), (20, 'jazz')]
+    """
+    inverse_list = []
+    cool_dict =[]
+    for key in (dictionary):
+        t=(dictionary[key],key)
+        inverse_list.append(t)
+    for i in inverse_list:
+        if len(i[1]) > b:
+            cool_dict.append(i)
+    new_list = sorted(cool_dict)
+    #print(new_list[len(cool_dict)-n:])
+    return new_list[len(cool_dict)-n:]
+
+#determine the words that are unique to each text (and that are the most common words)
+def unique_words(a,b,c,d):
+    """Examples:
+    >>> unique_words(['a','cat','dog'],['a','elephant','fish'],['fish','pet','car'],['a','truck','car'])
+    (['cat', 'dog'], ['elephant'], ['pet'], ['truck'])
+    """
+    a_list_words=[]
+    b_list_words=[]
+    c_list_words = []
+    d_list_words = []
+    for i in a:
+        if i not in b:
```

Comment:
This series of if statement can be shortened using boolean operation. `if i not in b and i not in c and i not in d` are equivalent statement.